src/message-queues.ts

```diff
@@ -208,8 +215,12 @@ export abstract class MessageQueue {
     });
     this.numPendingRequests++;
     this.numInFlightRequests++;
+    this.bytes += Buffer.byteLength(message.ackId, 'utf8');
 
-    if (this._requests.length >= maxMessages!) {
+    if (
+      this._requests.length >= maxMessages! ||
+      this.bytes >= MAX_BATCH_BYTES
```

Comment:
Does adding a large last message mean MAX_BATCH_BYTES could be exceeded by the grace size (10 bytes)?

Reply:
Ah, yeah, that's a good catch.